projects/Mallard/src/services/remote-config.ts

```diff
@@ -31,48 +29,35 @@ const configValues = {
 }
 
 class RemoteConfigService implements RemoteConfig {
-    private initialized: boolean
-
-    constructor() {
-        this.initialized = false
-    }
-
-    private setInitialized(isInitialized: boolean) {
-        this.initialized = isInitialized
-    }
-
     init() {
         remoteConfig()
             .setDefaults(remoteConfigDefaults)
             .then(() => remoteConfig().setConfigSettings(configValues))
             .then(() => {
-                remoteConfig().fetchAndActivate()
-                this.setInitialized(true)
-                console.log('Remote config fetched & activated!')
-                if (__DEV__) console.log(remoteConfig().getAll())
+                remoteConfig()
+                    .fetchAndActivate()
+                    .then(activated => {
+                        if (activated) {
+                            console.log('Remote config fetched & activated!')
+                            if (__DEV__) console.log(remoteConfig().getAll())
+                        } else {
+                            console.log('Remote config NOT activated!')
+                        }
+                    })
             })
             .catch(() => {
-                this.setInitialized(false)
                 console.log(
                     'Remote config not activated - something went wrong',
                 )
             })
     }
 
     getBoolean(key: RemoteConfigProperty): boolean {
-        if (this.initialized) {
-            return remoteConfig().getValue(key).value as boolean
-        }
-
-        return false
+        return remoteConfig().getValue(key).value as boolean
```

Comment:
Does this now mean if it fails to intiialise and this line fails, we will get errors in the app?

Reply:
No, for the first time, if it fails to fetch then it will return from defaults. Once it has fetched and then fails to fetch then it will return from last fetched values.